src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -124,12 +124,22 @@ public synchronized void launch(SlaveComputer computer, TaskListener listener) {
                     .stream().filter(s -> StringUtils.isNotBlank(s)).findFirst().orElse(null);
             slave.setNamespace(namespace);
 
+
+            TaskListener runListener = template.getListener();
+
             LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", new Object[] { namespace, podName });
-            pod = client.pods().inNamespace(namespace).create(pod);
+            try {
+                pod = client.pods().inNamespace(namespace).create(pod);
+            } catch (KubernetesClientException e) {
+                if (e.getMessage() != null && e.getMessage().contains("pod rejected due to") && e.getMessage().contains("openpolicyagent")) {
+                    runListener.getLogger().printf("ERROR: Unable to create pod. " + e.getMessage());
+                    PodUtils.cancelInvalidPodTemplateJob(pod, "OPA Violation");
```

Comment:
OPA is just one reason for potential pod rejection. The code should be refactored to handle any failure to create pod.

Reply:
I'm not aware of the full list of reasons for possible pod rejection, not even sure if such an extensive list can be compiled. Therefore, would a simple retry be the solution for some of these pod rejection reasons, therefore we wouldn't want them all to be lumped into the same block of canceling the caller build item?

Reply:
Does API server return a unique return code indicating admission controller rejections? In that case this code snippet can be refactored to look for that code and cancel the job as the job will never progress.

Reply:
Based on couple of examples of bad pod yaml's I tried, I see kube-apiserver returning code in the range of 400 - 499 for bad requests.. can we look for this and mark the build as failed?

```json
{
    "kind": "Status",
    "apiVersion": "v1",
    "metadata": {},
    "status": "Failure",
    "message": "admission webhook \"mutating-webhook.openpolicyagent.org\" denied the request: pod rejected due to unauthorized usage of spec.volumes[_].hostPath for paths [\"/var/run/docker.sock\"]",
    "reason": "pod rejected due to unauthorized usage of spec.volumes[_].hostPath for paths [\"/var/run/docker.sock\"]",
    "code": 400
}
```

```JSON
{
    "kind": "Status",
    "apiVersion": "v1",
    "metadata": {},
    "status": "Failure",
    "message": "Pod \"foo\" is invalid: spec.containers[0].image: Required value",
    "reason": "Invalid",
    "details": {
        "name": "foo",
        "kind": "Pod",
        "causes": [
            {
                "reason": "FieldValueRequired",
                "message": "Required value",
                "field": "spec.containers[0].image"
            }
        ]
    },
    "code": 422
}
```

Reply:
@narayanan My implementation already accounts for OPA violations with the search for `pod rejected due to` and `openpolicyagent` Strings, anything else I think would be too specific to our use case. 

What @Vlatombe is talking about is expanding the search for non-OPA violation reason pod rejections. But I don't know if they would all require canceling the originating build, or if pod failure and shortly a retry by the kubernetes plugin to create the pod again would be suffice.

Reply:
I think catching the 4xx errors should be enough. I'm thinking about any other implementation of validating webhook that could lead the api server to reject a pod spec

Reply:
4xx should fail the build, while 5xx should be retried.

Reply:
Updated the catch block to check the exception code instead of the message. Please take a look.